### PR TITLE
Add note to GPG key response if user has no keys

### PR DESCRIPTION
**Pull request by KN4CK3R**
~~Prevents the [invalid reponse](https://gitea.com/KN4CK3R.gpg) if there are no keys:~~
```
-----BEGIN PGP PUBLIC KEY BLOCK-----


=twTO
-----END PGP PUBLIC KEY BLOCK-----
```

~~Additional questions:~~
- ~~Should `ShowSSHKeys` do the same?~~
- ~~Is this an "api" endpoint and the response should be just the plain 404 response code instead of the html page? If yes we need to add an addition route for `/orgs/<org>.gpg` to get the same behaviour there too.~~

Adds a note if user has no keys.

### CODE REVIEW

**Review by 6543 — APPROVED** [review]

**Review by Gusted — APPROVED** [review]

**Comment by Gusted** [comment]
> * Should `ShowSSHKeys` do the same?

Yes.



> Is this an "api" endpoint and the response should be just the plain 404 response code instead of the html page? If yes we need to add an addition route for `/orgs/<org>.gpg` to get the same behaviour there too.

So far I can see, it's only used in the web part.

**Comment by KN4CK3R** [comment]
> So far I can see, it's only used in the web part.

Yes, but the response looks like something a tool uses as input which could be confused by html error page.

**Comment by Gusted** [comment]
> > So far I can see, it's only used in the web part.
> 
> Yes, but the response looks like something a tool uses as input which could be confused by html error page.

Well, I hope that tools can detect a simple 404 statuscode 😅 .

**Review by zeripath — APPROVED** [review]

File: routers/web/user/home.go [hunk 1]
```diff
@@ -736,6 +736,11 @@ func ShowGPGKeys(ctx *context.Context, uid int64) {
 		ctx.ServerError("ListGPGKeys", err)
 		return
 	}
+	if len(keys) == 0 {
+		ctx.NotFound("", nil)
```
Comment (zeripath) [inline routers/web/user/home.go]: Actually - no this should be:
```
-----BEGIN PGP PUBLIC KEY BLOCK-----
Note: This user hasn't uploaded any GPG keys.


=twTO
-----END PGP PUBLIC KEY BLOCK-----
```

**Review by zeripath — CHANGES_REQUESTED** [review]

**Comment by zeripath** [comment]
This should be a blank page.

**Review by Gusted — APPROVED** [review]

**Comment by Gusted** [comment]
CI errors are related `TestExportUserGPGKeys` test needs to be adjusted.

**Comment by Gusted** [comment]
Should this be backported?

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
This is very minor - a backport is reasonable but equally not sending one is also reasonable.

**Pull request merged**